packages/@lwc/engine-core/src/framework/component.ts

```diff
@@ -21,13 +21,16 @@ const signedTemplateMap: Map<LightningElementConstructor, Template> = new Map();
  * will prevent this function from being imported by userland code.
  */
 export function registerComponent(
-    Ctor: LightningElementConstructor,
+    // We typically expect a LightningElementConstructor, but technically you can call this with anything
+    Ctor: any,
```

Comment:
Not an issue in this simple function, but `unknown` is better for things that need to be type-checked.